### PR TITLE
Update genomics software

### PR DESCRIPTION
**Pull request by mschubert**

### CODE REVIEW

**Comment by mmokrejs** [comment]
Hi @mschubert,
  it is a pity the automated checks failed due to other packages (not touched by you) but I cannot really help with obeying that. Let's hope @jlec or @SoapGentoo will merge them despite errors.

**Comment by SoapGentoo** [comment]
@mschubert could you please send these to the main repo instead? The overlay is a barren wasteland of extremely low-quality packages that just rot away and aren't tested by our tinderbox. https://github.com/gentoo/gentoo/pulls

File: sci-biology/STAR/STAR-2.5.3a.ebuild [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# sci-biology/htslib
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+# contains bundled STAR-Fusion
+src_prepare(){
+	sed -e "s/= gcc/= $(tc-getCC)/;s/-O2/${CFLAGS}/g" -i source/htslib/Makefile || die
+	sed -e "s/-O3/${CFLAGS}/g" -i source/Makefile || die
+}
```
Comment (SoapGentoo) [inline sci-biology/STAR/STAR-2.5.3a.ebuild]: I strongly doubt you even tested this ebuild...

**Comment by mmokrejs** [comment]
@mschubert  maybe try to inspect/take over changes to kaldi from https://github.com/gentoo/sci/pull/826 as well?

**Comment by mschubert** [comment]
Thanks for the comments, I'll resend to the main tree.

File: sci-biology/STAR/STAR-2.5.3a.ebuild (hunk 1 above)
Comment (mschubert) [reply sci-biology/STAR/STAR-2.5.3a.ebuild]: The ebuild is functional, I have it installed on my system. The commits, however, were only a EAPI and version bump (as referred to in the commit message), I did not check whether these lines are required or even functional.

If this is something I should have investigated more carefully, please write this during review and do not assume I haven't tested it.

Comment (SoapGentoo) [reply sci-biology/STAR/STAR-2.5.3a.ebuild]:
```
$ ebuild STAR-2.5.3a.ebuild compile
[...]
 * ERROR: sci-biology/STAR-2.5.3a::science failed (prepare phase):
 *   eapply_user (or default) must be called in src_prepare()!
 * 
 * Call stack:
 *            ebuild.sh, line  776:  Called __ebuild_main 'prepare'
 *   phase-functions.sh, line 1053:  Called __dyn_prepare
 *   phase-functions.sh, line  383:  Called die
 * The specific snippet of code:
 *   		die "eapply_user (or default) must be called in src_prepare()!"
 * 
 * If you need support, post the output of `emerge --info '=sci-biology/STAR-2.5.3a::science'`,
 * the complete build log and the output of `emerge -pqv '=sci-biology/STAR-2.5.3a::science'`.
 * The complete build log is located at '/var/tmp/portage/sci-biology/STAR-2.5.3a/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-biology/STAR-2.5.3a/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-biology/STAR-2.5.3a/work/STAR-2.5.3a'
 * S: '/var/tmp/portage/sci-biology/STAR-2.5.3a/work/STAR-2.5.3a'
```

Comment (SoapGentoo) [reply sci-biology/STAR/STAR-2.5.3a.ebuild]: In general it goes without saying that if you add a new ebuild to the system, you have at least done the basic `ebuild ... install` routines to check that the basics of it works.

Comment (mschubert) [reply sci-biology/STAR/STAR-2.5.3a.ebuild]: Yes, you're right, this now fails. - I'm guessing this check has been added to portage more recently than my PR? (and prefix portage at the time).

I'll fix it before resubmitting. Thank you for spotting this.

Comment (SoapGentoo) [reply sci-biology/STAR/STAR-2.5.3a.ebuild]: Nope, that `eapply_user()` check has been there for ages (since EAPI 6 support was integrated into Portage).

**Comment by SoapGentoo** [comment]
@mschubert don't worry, I will add them to the main tree (some of those ebuilds need to be unbundled, such as STAR).

**Comment by SoapGentoo** [comment]
@mschubert HTseq, STAR and kallisto are in the main tree now. Salmon needs quite a bit of work, as it has a ton of dependencies that will be automagically downloaded. This is against Gentoo's QA policies. Please add those dependencies separately and open a new PR for Salmon against the main tree.

**Pull request closed without merging**